numpy/core/src/umath/reduction.c

```diff
@@ -331,15 +331,11 @@ PyUFunc_ReduceWrapper(PyArrayObject *operand, PyArrayObject *out,
         goto fail;
     }
 
-    /* Start with the floating-point exception flags cleared */
-    npy_clear_floatstatus_barrier((char*)&iter);
```

Comment:
The move here is actually intentional: I figure that the initial copy could (at least currently) in theory set floating point warning flags.